src/toil/common.py

```diff
@@ -1294,32 +1304,78 @@ def getToilWorkDir(configWorkDir: Optional[str] = None) -> str:
         return workDir
 
     @classmethod
-    def get_toil_coordination_dir(cls, configWorkDir: Optional[str] = None) -> str:
+    def get_toil_coordination_dir(cls, config_work_dir: Optional[str], config_coordination_dir: Optional[str]) -> str:
         """
         Return a path to a writable directory, which will be in memory if
         convenient. Ought to be used for file locking and coordination.
 
-        :param configWorkDir: Value passed to the program using the --workDir flag
-        :return: Path to the Toil coordination directory.
+        :param config_work_dir: Value passed to the program using the
+               --workDir flag
+        :param config_coordination_dir: Value passed to the program using the
+               --coordinationDir flag
+
+        :return: Path to the Toil coordination directory. Ought to be on a
+                 POSIX filesystem that allows directories containing open files to be
+                 deleted.
         """
 
-        # Get our user ID
-        user_id = os.getuid()
-        in_memory_base = os.path.join('/var/run/user', str(user_id), 'toil')
-        if os.path.exists(in_memory_base):
-            # We have an in-memory directory to use
-            return in_memory_base
-        else:
-            try:
-                # Try to make it opportunistically.
-                os.makedirs(in_memory_base, exist_ok=True)
-                return in_memory_base
-            except:
-                pass
+        # Define higher-order functions over Optional because I learned Haskell
+        # once.
 
-        # Otherwise use the on-disk one.
-        return cls.getToilWorkDir(configWorkDir)
+        T = TypeVar('T')
+        def omap(o: Optional[T], f: Callable[[T], T]) -> Optional[T]:
+            """
+            Map a function over an optional.
+            """
+            return f(o) if o is not None else None
 
+        def ofilter(o: Optional[T], p: Callable[[T], bool]) -> Optional[T]:
+            """
+            Filter an optional with a predicate function.
+            """
+            return None if (o is None or not p(o)) else o
+
+        def try_path(path: str) -> Optional[str]:
```

Comment:
This function in particular, but also possibly omap and ofilter could probably live in lib and eventually be reused.  It's definitely something I could see reusing.